### ISSUE TITLE
100% ANR

### ISSUE THREAD

**Issue by DmitriyZaitsev** [koral--/android-gif-drawable#108]
Hi koral,
I use *GifDrawable* with *GifImageView* in *ListView*. I found ANR happens with 100% probability after ~1 minute of continuous fast scrolling even with manual recycling.
Here is part of my code inside adapter's getView():
```java
//...
Drawable drawable = gifView.getDrawable();
holder.gifView.setImageDrawable(null);  // controversial
if (drawable instanceof GifDrawable) {
     ((GifDrawable) drawable).recycle();
}
drawable = new GifDrawable(path);
holder.gifView.setImageDrawable(drawable);
//...
```
What can cause the ANR? Or maybe I do smth wrong?

**Comment by DmitriyZaitsev** [comment]
I use the latest version, definitely.

**Comment by koral--** [comment]
OK, I'll try to reproduce that.

**Comment by koral--** [comment]
I could not reproduce ANR while continuous scrolling but have 2 ideas what may help:
1. using one executor for all drawables inside list instead of separate executor for each instance (which is default behavior), setting custom executor is possible through `GifDrawableBuilder#taskExecutor()`. Doing so will avoid creation hundreds of threads which is some overhead even if most of them are not working because their drawables are not visible.
2. suspend animation while scrolling (#73)

With current snapshot you can try approach 1. The second will be added soon.

**Comment by DmitriyZaitsev** [comment]
Before, to fix this I simply used ```GifDrawable#mExecutor``` as ```static```.
Today I checked the ```1.1.2-SNAPSHOT``` and saw that you did almost the same but in different more flexible way.
Seems to be fixed.
Thanks.

**Comment by DmitriyZaitsev** [comment]
One more question. Is code below correct (I mean using of ```GifDrawableBuilder#with()``` and ```GifDrawableBuilder#from()```)?

```java
final Drawable oldDrawable = gifView.getDrawable();

if (mGifBuilder == null) {
    mGifBuilder = new GifDrawableBuilder().threadPoolSize(5);
}

try {
    mGifBuilder.from(path);

    if (oldDrawable instanceof GifDrawable) {
        mGifBuilder.with((GifDrawable) oldDrawable);
    }

    gifView.setImageDrawable(mGifBuilder.build());
} catch (final IOException e) {
    gifView.setImageDrawable(null);
}
```

**Comment by koral--** [comment]
Your usage should be correct, but it is not (yet) working. Artifacts happen because at the time of creation new instance with reusing, rendering of the frame of old instance may be in progress and the last "old" modification may occur after new one is created.
I'll fix this ASAP.

**Comment by koral--** [comment]
Artifacts should not appear now. Fixed in v 1.1.2.
